pkg/jobstore/state_updater.go

```diff
@@ -0,0 +1,21 @@
+package jobstore
+
+import (
+	context "context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/util"
+	"github.com/rs/zerolog/log"
+)
+
+func MakeEvaluationStateUpdater(ctx context.Context, store Store) func(*models.Evaluation) {
+	detachedContext := util.NewDetachedContext(ctx)
```

Comment:
Mostly because we want to capture the store, and don't want to have to pass it in with each call.  This way, the eval broker doesn't need to know anything about the store, it can work with a simpler interface of `func(*models.Evaluation)`